### PR TITLE
Adds hooks to login modals for integration tests

### PR DESCRIPTION
**Pull request by zephraph**
This adds hooks to our auth modals for more reliable detection during integration testing. 

<details>
  <summary>📦 Published PR as canary version: <code>25.25.0-canary.3234.53369.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```sh
  npm install @artsy/reaction@25.25.0-canary.3234.53369.0
  # or 
  yarn add @artsy/reaction@25.25.0-canary.3234.53369.0
  ```
</details>

### CODE REVIEW

File: src/Components/Authentication/Desktop/LoginForm.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class LoginForm extends Component<FormProps, LoginFormState> {
           }
 
           return (
-            <Form onSubmit={handleSubmit}>
+            <Form onSubmit={handleSubmit} data-test-login-form>
```
Comment (damassi) [inline src/Components/Authentication/Desktop/LoginForm.tsx]: We'll want to follow [this format](https://github.com/damassi/reaction/blob/d43843bf504b980fa8d6862cc3520c39b9c6065e/src/Apps/Order/Components/PaymentPicker.tsx#L270)

Comment (zephraph) [reply src/Components/Authentication/Desktop/LoginForm.tsx]:
```suggestion
            <Form onSubmit={handleSubmit} data-test="LoginForm">
```

File: src/Components/Authentication/Mobile/LoginForm.tsx [hunk 2]
```diff
@@ -128,7 +128,7 @@ class MobileLoginFormWithSystemContext extends Component<
           const { currentStep, isLastStep } = wizard
 
           return (
-            <MobileContainer>
+            <MobileContainer data-test-login-form>
```
Comment (zephraph) [inline src/Components/Authentication/Mobile/LoginForm.tsx]:
```suggestion
            <MobileContainer data-test="LoginForm">
```

**Comment by artsyit** [comment]
:rocket: PR was released in `v25.25.0` :rocket:

**Pull request merged**